packages/cumulus-api-client/CumulusApiClient.js

```diff
@@ -131,30 +137,65 @@ class CumulusApiClient {
     }
   }
 
+  /**
+   * Calls create token, then updates the cache with the new record
+   * @memberof CumulusApiClient
+   *
+   * @throws {CumulusAuthTokenError} - throws on any error
+   * @returns {string} - returns active valid bearer token
+   */
+  async _createAndUpdateNewAuthToken() {
+    try {
+      const updateToken = await this.createNewAuthToken();
+      await this._updateAuthTokenRecord(
+        updateToken
+      );
+      return updateToken;
+    } catch (error) {
+      throw new CumulusAuthTokenError(error);
+    }
+  }
+
+  /**
+   * Updates the cached bearer token and sets cacheInitialized to true
+   * @memberof CumulusApiClient
+   *
+   * @returns {string} -
+   */
+  async _initializeCache() {
+    this.logger.info('Creating new token on class creation');
+    const token = await this._createAndUpdateNewAuthToken();
+    this.cacheInitialized = true;
+    return token;
+  }
+
   /**
    * Gets a token using the CumulusApiClient caching scheme:
-   *  1) Attempts to get a valid token from the cache table
+   *  1) If the class hasn't been initialized, cretes a new token *else*
+   *     attempts to get a valid token from the cache table.
    *  2) If it is successful, check (if possible) if the token is about to expire
    *  3) If the token does not exist, or is about to expire, generate a new token
    *     and store it inthe cache
+   * @memberof CumulusApiClient
    *
    * @returns {string} - returns active valid bearer token
    */
   async getCacheAuthToken() {
-    let token;
     try {
-      token = await this._getAuthTokenRecord();
+      let token;
```

Comment:
This is the actual logic change, the new methods above are to DRY up/ensure updates on create calls.